### DISCUSSION TITLE
Call EntityPlaceEvent for spawn eggs

### DISCUSSION THREAD

**Discussion by NonSwag** [PaperMC/Paper#11564] [Ideas]
### Is your feature request related to a problem?

EntityPlaceEvent is only called for armor stands, boats, minecarts and end crystals
item frames and paintings have its own event (HangingPlaceEvent)
but spawn eggs are not covered by any event

### Describe the solution you'd like.

Fire the EntityPlaceEvent for spawn eggs as well

### Describe alternatives you've considered.

Intercepting the PlayerInteractEvent

**Comment by Leguan16** [comment]
So you want the EntityPlaceEvent to be called for spawn eggs because you need to know which player placed the egg?

**Comment by Doc94** [comment]
This sounds like https://github.com/PaperMC/Paper/discussions/10527 and https://github.com/PaperMC/Paper/issues/6487 ?

**Comment by electronicboy** [comment]
PlayerLaunchProjectileEvent is used for eggs, the EntityPlaceEvent is fired in cases where an interaction directly spawns an entity into the world, and that's not the case here

**Reply by NonSwag** [reply]
> PlayerLaunchProjectileEvent is used for eggs, the EntityPlaceEvent is fired in cases where an interaction directly spawns an entity into the world, and that's not the case here

I don't think I can follow.
Placing a spawn egg does directly spawn the corresponding entity in the world, or am I wrong?

**Reply by electronicboy** [reply]
I, er, apparently brainfarted and got the wrong egg

**Comment by Leguan16** [comment]
Also: https://github.com/PaperMC/Paper/pull/7377 might be what we are looking for here.
